src/components/SurveyAnswer/ChoiceAnswer/MultiplePickAnswer.tsx

```diff
@@ -0,0 +1,74 @@
+import Image from 'next/image';
+import SimpleBar from 'simplebar-react';
+
+import { SurveyAnswerProps } from '..';
+
+import styles from './ChoiceAnswer.module.scss';
+import 'simplebar-react/dist/simplebar.min.css';
+
+const MultiplePickAnswer = ({
+  question,
+  currentResponse,
+  onResponseChange,
+}: SurveyAnswerProps) => {
+  const onAnswerSelect = (answerId: string) => {
+    const currentAnswers = currentResponse?.answers || [];
+    const isAnswerSelected = currentAnswers.some(({ id }) => id === answerId);
+
+    if (isAnswerSelected) {
+      const updatedAnswers = currentAnswers.filter(({ id }) => id !== answerId);
+
+      if (updatedAnswers.length === 0) {
+        return onResponseChange(null);
+      }
+
+      onResponseChange({
+        questionId: question.id,
+        answers: updatedAnswers,
+      });
+    } else {
+      onResponseChange({
+        questionId: question.id,
+        answers: currentAnswers.concat({ id: answerId }),
+      });
```

Comment:
```suggestion
        return setAnswers(null);
      }

      setAnswers(updatedAnswers);
    } else {
      setAnswers(currentAnswers.concat({ id: answerId }));
```

As we are not in a `useEffect` hook, we can easily extract code into methods. What about adding a `setAnswers` method that would call the `onResponseChange`. 

This would make the code easier to read IMO :) I also needed some time to remember that `onResponseChange` is a handler set from the parent component to update the answers for a given question ^^

Reply:
Updated in [d80b942](https://github.com/hanam1ni/react-surveyor/pull/97/commits/d80b942a0e32bf194dea9f5fe6ebeb172a7b95be)